app/broadcaster.py

```diff
@@ -7,6 +7,22 @@
 import base64
 from status import Status
 
+class HTTPBasicThenDigestAuth(requests.auth.HTTPDigestAuth):

```

Comment:
https://github.com/kennethreitz/requests/blob/master/requests/auth.py
http://docs.python-requests.org/en/master/user/authentication/

Yes, I basically reimplemented `curl --anyauth`, which tries basic auth then digest auth (then other auths but that is not needed).

Some cameras are basic auth only, some others are digest auth only, some are both.